content/api/commands/attachfile.md

```diff
@@ -0,0 +1,235 @@
+
+### On a hidden input
+
+```javascript
+cy.get('input[type=file]').attachFile('file.json', { force: true })
+```
+
+In many cases in modern applications, the underlying file input is hidden from view, and activated by a user clicking on a button. In these cases, you will need to tell Cypress to ignore its actionability checks and attach the file even though a user would not be able to directly activate the file input.
+
+### From a fixture
+```javascript
+cy.fixture('file.json', { encoding: null }).as('myFixture')
+cy.get('input[type=file]').attachFile('@myFixture')
+```
+
+Note the use of `null` encoding. By default, `cy.fixture()` and `cy.readFile()` attempt to interpret files read from disk, which would result in a JSON file being decoded and re-encoded as a utf-8 string - the contents would be preserved, but formatting would not be and the encoding might change. See [`cy.fixture`](/api/commands/fixture) or [`cy.readFile`](/api/commands/readfile) for more details on file encoding.
+
+### From an API response
+
+```javascript
+cy.request('http://localhost:8888/users/827')
+  .its('body')
+  .as('responseBody')
+
+cy.get('input[type=file]').attachFile('@responseBody')
+```
+
+### Processing data inside the test
+
+```javascript
+cy.readFile('users.json').then((users => {
+  users[0].username = 'JohnCena'
+})).as('myFile')
+
+cy.get('input[type=file]').attachFile('@myFile')
+```
+
+### Attaching multiple files
+
+```javascript
+cy.get('input[type=file]').attachFile([
+  'file1.json',
+  'file2.json',
+  'file3.json',
+])
+```
+
+<Alert type="warning">
+
+This will fail unless the file input has the `multiple` property.
+
+</Alert>
+
+### Attaching a file with custom filename and lastModified
+
+```javascript
+cy.get('input[type=file]').attachFile({
+  contents: 'path/to/file.json',
+  fileName: 'users.json',
+  lastModified: new Date('Feb 18 1989').valueOf()
+})
+```
+
+### Dropping a file on the document
+
+```javascript
+cy.document().attachFile('file.json', { action: 'drag-n-drop' })
+```
+
+## Notes
+
+### Existence
+
+#### Default file existence assertion
+
+When passed a file path, `.attachFile()` asserts that the file exists and will fail if it does not exist. It will retry reading the file if it does not initially exist until the file exists or the command times out.
+
+```javascript
+// will fail after the defaultCommandTimeout is reached
+cy.get('input[type=file]').attachFile('does-not-exist.yaml')
+```
+
+### Actionability
+
+#### The element must first reach actionability
+
+`.attachFile()` is an "action command" that follows all the rules
+[defined here](/guides/core-concepts/interacting-with-elements).
+
+## Rules
+
+### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
+
+<List><li>`.attachFile()` requires being chained off a command that yields DOM
```

Comment:
Can these be markdown lists?

Reply:
I am not exactly sure why they're not, but I was copying the pattern from other files in this directory - they universally use `<List>` in their Rules section. Without understanding Chesterton's fence, I didn't want to change it. I'm wondering if maybe some of our custom scripts rely on it?

![image](https://user-images.githubusercontent.com/3003404/146043293-ac22fc7d-7348-4df7-abce-fe33296defa2.png)

Reply:
I'm not entirely sure what's up with `<List>`, but the `List` component is just this:

```vue
<template>
  <ul>
    <slot />
  </ul>
</template>
```

I loaded all the commit diffs containing `<List` into vscode with:

```sh
git log -G'<List' -p | code -
```

and then I searched for `(?<![</])\bList\b` to try to find any references to `List` that aren't preceded by `<` or `/` (to filter out `<List>` and `</List>` and found nothing of interest.

I think we're ok making it a markdown list.